TypeCobol/Compiler/AntlrUtils/ParserDiagnosticErrorListener.cs

```diff
@@ -65,10 +65,15 @@ public class ParserDiagnostic : Diagnostic
             // TO DO - IMPORTANT : this is the INITIAL line number, and not the CURRENT line number
             // This is enough to pass all unit tests, but will return false informations in real usage !
             // for real line number, use a Snapshot
-            if (Line < 0 && OffendingSymbol != null)
+            if (OffendingSymbol != null)
             {
-                CodeElement e = OffendingSymbol as CodeElement;
-                if (e != null && e.ConsumedTokens.Count > 0) Line = e.ConsumedTokens[0].Line;
+                var copyToken = (offendingSymbol as ImportedToken)?.CopyDirective?.COPYToken;
+                if (copyToken != null) Line = copyToken.Line;
```

Comment:
Why use offendingSymbol when OffendingSymbol was used for non-null test?

Reply:
done : use of only OffendingSymbol.
https://github.com/TypeCobolTeam/TypeCobol/pull/1602/commits/c75e93cbbbe150f7b2561c1696660261a82adfbe